### PR TITLE
[21.09] Fix tool id update when updating tool shed tools in editor

### LINKED ISSUE #13153: When tool version is changed in the workflow editor, tool_id is unchanged in downloaded .ga file
**Describe the bug**
If I change the version of a tool used in a workflow and then download the workflow, the tool_id and content_id show the previous tool version.  Only the changeset_revision is updated for the workflow step.

**Galaxy Version and/or server at which you observed the bug**
21.09

**To Reproduce**
1. On Galaxy Europe or Galaxy Australia upload the Galaxy 101 for everyone tutorial
`https://training.galaxyproject.org/training-material/topics/introduction/tutorials/galaxy-intro-101-everyone/workflows/main_workflow.ga`
2. Edit the workflow and change the version of Scatterplot with ggplot2 from '2.2.1+galaxy1' to '2.2.1+galaxy2'
3. Download the workflow file
4. See that the tool_id and content_id fields for the ggplot2 steps are `toolshed.g2.bx.psu.edu/repos/iuc/ggplot2_point/ggplot2_point/2.2.1+galaxy1`

**Expected behavior**
Tool version changes in tool_id and content_id fields when the version is changed

**Additional context**
It all still works, since the changeset revision is updated when the tool version is updated through the workflow editor, so if the workflow is downloaded and reuploaded the version change is preserved, but having tool_id and content_id showing the previous version in the .ga file is confusing.

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes #13153, likely broken in https://github.com/galaxyproject/galaxy/pull/11898#discussion_r787691658

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: test/integration_selenium/test_workflow_repository_tool_update.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        editor = self.components.workflow_editor
+        editor.node._(label="compose_text_param").wait_for_and_click()
+        editor.tool_version_button.wait_for_and_click()
+        assert self.select_dropdown_item('Switch to 0.1.1'), 'Switch to tool version dropdown item not found'
+        self.screenshot("workflow_editor_version_update_tool_shed")
+        self.sleep_for(self.wait_types.UX_RENDER)
+        self.assert_workflow_has_changes_and_save()
+        workflow = self.workflow_populator.download_workflow(workflow_id)
+        assert workflow['steps']['0']['tool_version'] == '0.1.1'
+        assert workflow['steps']['0']['tool_id'] == 'toolshed.g2.bx.psu.edu/repos/iuc/compose_text_param/compose_text_param/0.1.1'
```
Comment (jmchilton) [inline test/integration_selenium/test_workflow_repository_tool_update.py:39]: This test is beautiful.

**Review by jmchilton — APPROVED** [review]

File: client/src/components/Workflow/Editor/Node.vue [hunk 2]
```diff
@@ -314,6 +314,7 @@ export default {
         },
         setData(data) {
             this.config_form = data.config_form;
+            this.content_id = data.config_form?.id || data.content_id || this.content_id;
```
Comment (guerler) [inline client/src/components/Workflow/Editor/Node.vue]: Thanks a lot @mvdbeek. You identified the correct issue. We could alternatively fix this by consistently returning the `content_id` within the API response like shown here (as we probably should): https://github.com/galaxyproject/galaxy/commit/c46653cdd575725f5243b7c8532b7ae003f00037.
That would avoid attempting to deal with corner cases, and the fact that the `content_id` is provided in `data` upon initialization but not when updating the module state.

Comment (mvdbeek) [reply client/src/components/Workflow/Editor/Node.vue]: Yeah, I think that's consistent, I'll pick that commit

Comment (mvdbeek) [reply client/src/components/Workflow/Editor/Node.vue]: Hmm, that didn't quite work https://github.com/galaxyproject/galaxy/runs/4868498354?check_suite_focus=true#step:7:2323

**Review by guerler — APPROVED** [review]

**Pull request merged**